src/MobileRedirection.php

```diff
@@ -64,27 +64,30 @@ public function register() {
 		add_filter( 'amp_default_options', [ $this, 'filter_default_options' ] );
 		add_filter( 'amp_options_updating', [ $this, 'sanitize_options' ], 10, 2 );
 
-		$is_mobile_redirect_enabled = AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT );
-		$sandboxing_level           = amp_get_sandboxing_level();
+		if ( ! amp_is_canonical() ) {
+			$sandboxing_level           = amp_get_sandboxing_level();
+			$is_mobile_redirect_enabled = AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT );
 
-		// Add alternative link if mobile redirection is enabled or sandboxing level is set to loose or moderate.
```

Comment:
Could you restore this comment?

Reply:
Thanks for pointing that out. Restored it in https://github.com/ampproject/amp-wp/pull/7426/commits/2c7b186347fbc0ba76f271d83feb36644f0c4a07